### DISCUSSION TITLE
Coping with missing field

### DISCUSSION THREAD

**Discussion by hansbogert** [cue-lang/cue#3276] [Q&A]
I have the following situation:

```cue
foo: fieldA: {bar: "1"}
foo: fieldAB: {bar: "1"}

foo: A = [N = =~".*[^B]"]: {
        _musthaveaB: foo["\(N)B"].bar == A.bar & true
}
``` 
This works fine. Changing the second line to:

```
foo: fieldAB: {bar: "2"}
```

Results in:

```
 cue vet foo.cue
foo.fieldA._musthaveaB: conflicting values true and false:
    ./foo.cue:4:24
    ./foo.cue:5:15
    ./foo.cue:5:41
```
Still as expected.

However, if we comment out the second line, `cue vet` succeeds.

I thought the unification with true was pretty elegant in my case, but only after a while  noticed this false positive.

What would be the idiomatic way to check that every field not ending in "B", should  have a `...B` counterpart with the same value? 

The above behaviour with a missing but expected field not showing an error after vet or evaluation has been the biggest gotcha with Cue so far.

**Comment by the-nurk** [comment]
alias `foo`? and use the alias?

**Reply by hansbogert** [reply]
How would that look like?

**Reply by the-nurk** [reply]
A=[pattern_constraint]: _ is just always going to be wrong if i'm not mistaken.

```
#B: string & =~".*[^B]"
F=foo: [N=#B]: {
        _musthaveaB: F["\(N)B"].bar == A.bar & true
}
```

?

**Reply by hansbogert** [reply]
the problem is not with aliasing, the problem is more semantically:

```
 _musthaveaB: F["\(N)B"].bar == A.bar & true
```

will not be an error even if the `F["\(N)B"].bar` will never be defined. The only way to make this an error is to reference it in a non-hidden field and demand concrete values by `cue eval -c` for example. However, that means the guard value of `musthaveaB` becomes part of the data which feels wrong.

**Reply by the-nurk** [reply]
i mean... am i missing something?

https://cuelang.org/play/?id=9vidvRv2rka#w=function&i=cue&f=eval&o=cue

yeah not sure what you're trying to accomplish as an end goal...
